### PR TITLE
Compat

### PR DESCRIPTION
**Pull request by iamed2**
Removed deps, so this will require a minor version bump

This should work on 0.7 now yay

### CODE REVIEW

**Comment by iamed2** [comment]
Requires https://github.com/JuliaLang/Compat.jl/pull/490

**Comment by iamed2** [comment]
What's with the tests passing but Travis saying they failed?

**Comment by iamed2** [comment]
Can reproduce locally. There's some weird stuff going on.

**Comment by iamed2** [comment]
Alright this 0.6 test failure is weird so I say we return to this once the Syslogs METADATA PR is merged and see if that helps.

**Comment by rofinn** [comment]
Maybe we should add `__precompile__(true)` to Syslogs.jl and see what happens?

File: test/runtests.jl [hunk 1]
```diff
@@ -12,17 +17,19 @@ files = [
...
 haskey(ENV, "MEMENTO_BENCHMARK") ? files = ["benchmarks.jl"] : nothing
 
-using Memento
+# for loggers.jl
+struct TestError <: Exception
+    msg
+end
 
-cd(dirname(@__FILE__))
+@testset "Memento" begin
```
Comment (rofinn) [inline test/runtests.jl]: I'm not a fan of having global testsets. Maybe we should switch to using TestSetExtensions.jl?

Comment (iamed2) [reply test/runtests.jl]: Yes I'll add it in tomorrow

**Pull request merged**